### PR TITLE
Windows: fix a crash when splitting command-line arguments

### PR DESCRIPTION
**Pull request by adriansr**
This fixes a crash when fetching a process arguments. The command-line string read from the target process memory sometimes is not terminated.
It can cause bogus characters to be appended to the command-line or a crash.

### CODE REVIEW

File: sys/windows/syscall_windows.go [hunk 1]
```diff
@@ -503,9 +503,17 @@ func GetUserProcessParams(handle syscall.Handle, pbi ProcessBasicInformation) (p
 	return params, nil
 }
 
+// ReadProcessUnicodeString returns a zero-terminated UTF-16 string from another
+// process memory.
```
Comment (andrewkroh) [inline sys/windows/syscall_windows.go]:
```suggestion
// process's memory.
```

File: sys/windows/syscall_windows.go [hunk 2]
```diff
@@ -515,12 +523,26 @@ func ReadProcessUnicodeString(handle syscall.Handle, s *UnicodeString) ([]byte,
 	return buf, nil
 }
 
-// Use Windows' CommandLineToArgv API to split an UTF-16 command line string
-// into a list of parameters.
+// ByteSliceToStringSlice uses CommandLineToArgv API to split an UTF-16 command
+// line string into a list of parameters.
 func ByteSliceToStringSlice(utf16 []byte) ([]string, error) {
```
Comment (andrewkroh) [inline sys/windows/syscall_windows.go]: Could you please add a unit test that covers these cases (odd size, not null-terminated).
